core/app/models/spree/order.rb

```diff
@@ -154,7 +154,7 @@ def self.register_line_item_comparison_hook(hook)
 
     # For compatiblity with Calculator::PriceSack
     def amount
-      line_items.inject(0.0) { |sum, li| sum + li.amount }
+      line_items.map(&:amount).sum
```

Comment:
Ok, also keep it. This PR isn't for actually changing behaviour, and an additional roundtrip would be just that.